src/main/java/org/apache/sysds/runtime/controlprogram/federated/FederatedResponse.java

```diff
@@ -94,10 +94,11 @@ public void throwExceptionFromResponse() throws Exception {
 	 * If the map is empty, it means that no privacy constraints were found.
 	 * @param checkedConstraints map of checked constraints from the PrivacyMonitor
 	 */
-	public void setCheckedConstraints(ConcurrentHashMap<PrivacyLevel,LongAdder> checkedConstraints){
+	public void setCheckedConstraints(HashMap<PrivacyLevel,LongAdder> checkedConstraints){
 		if ( checkedConstraints != null && !checkedConstraints.isEmpty() ){
-			this.checkedConstraints = checkedConstraints;
-		}
+			this.checkedConstraints = new HashMap<PrivacyLevel, LongAdder>();
+			this.checkedConstraints.putAll(checkedConstraints);
+		}	
```

Comment:
question.
Are you leveraging that it is not concurrent? or that the likelihood of two elements having the same hash is low?

Reply:
It is not necessary for it to be concurrent. An enum should have the same hash value and it is unlikely to conflict with other enums. I could also have used EnumMap, but it is not necessary.

Reply:
EnumMap sounds good :) did not know that one, but as you say it is not necessary.

Reply:
Now I changed it anyway. The documentation says that it is efficient :smile: